### DISCUSSION TITLE
Failed to initiate the GitHub login process. Please try again.

### DISCUSSION THREAD

**Discussion by wwzhxbc** [community/community#17485] [Copilot Conversations]
![图片](https://user-images.githubusercontent.com/103719188/170912057-a758b0c2-4829-486f-977e-b069a6cd5cf5.png)
![图片](https://user-images.githubusercontent.com/103719188/171353766-37444f28-0eea-4bc9-8cd1-500f75fc98a8.png)
![图片](https://user-images.githubusercontent.com/103719188/171353796-4858d64e-5910-46b4-b511-471d316940b1.png)

**Comment by davecheney** [comment]
Which editor and version of the copilot plugin are you using?

**Reply by YoLung** [reply]
idea 2022.1.1(Ultimate Edition) + The latest version of copilot

**Reply by davecheney** [reply]
Is it possible that your employer, internet service provider, or government, may be intercepting your SSL traffic?

**Reply by jansorg** [reply]
Another possibility is that a proxy is needed on your system. The latest few releases contain technical changes, which don't fully support proxy servers yet.

**Reply by YoLung** [reply]
> 
Can I trouble you to adapt it as soon as possible, Chinese users really need this optimization。Thanks

**Reply by YoLung** [reply]
> Is it possible that your employer, internet service provider, or government, may be intercepting your SSL traffic?

It should be that the network is restricted, but I still can't log in after opening the proxy. I think it should be a plug-in adaptation problem, as the brother said above.

**Reply by RanY-Luck** [reply]
在以往的版本中没有出现这种错误

**Reply by luojiaheng81** [reply]
> > 
> 
> 能不能麻烦你尽快适配一下，中国用户真的很需要这个优化。谢谢

用回1.1.20就可以了

**Reply by YoLung** [reply]
> > > 
> > 
> > 
> > 能不能麻烦你尽快适配一下，中国用户真的很需要这个优化。谢谢
> 
> 用回1.1.20就可以了

1.1.20.1417？

**Reply by luojiaheng81** [reply]
> > > > 
> > > 
> > > 
> > > 能不能麻烦你尽快适配一下，中国用户真的很需要这个优化。谢谢
> > 
> > 
> > 用回1.1.20就可以了
> 
> 1.1.20.1417？

对

**Reply by YoLung** [reply]
> > > > > 
> > > > 
> > > > 
> > > > 能不能麻烦你尽快适配一下，中国用户真的很需要这个优化。谢谢
> > > 
> > > 
> > > 用回1.1.20就可以了
> > 
> > 
> > 1.1.20.1417？
> 
> 对

It's really strange. Yesterday I could not log in with version 1.1.20.1417, but I successfully logged in today. Thanks

**Reply by Tang200614** [reply]
Today, I cannot log in using version 1.1.20.1417, webstorm prompts me to update to the latest version, but this problem will occur again when I update to the latest version

**Reply by pakchoily** [reply]
> Today, I cannot log in using version 1.1.20.1417, webstorm prompts me to update to the latest version, but this problem will occur again when I update to the latest version

add api.github.com ip in your `hosts`，in Chinese，may be this reason。 [#16965 ](https://github.com/orgs/github-community/discussions/16965#discussioncomment-2881851)

**Reply by yagamiy1023** [reply]
> > 今天使用1.1.20.1417版本无法登录，webstorm提示我更新到最新版本，但是更新到最新版本又会出现这个问题
> 
> 在你的 api.github.com 中添加 ip `hosts`，中文，可能是这个原因。[#16965](https://github.com/orgs/github-community/discussions/16965#discussioncomment-2881851)
1.1.20.1417已经不支持了，只能更新版本了

**Reply by sscoo** [reply]
v2ray要在设置里打开允许局域网的连接

**Comment by wwzhxbc** [comment]
Hello, can you tell me which versions support proxy server?



------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "github/feedback"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年5月30日(星期一) 下午3:38
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [github/feedback] Failed to initiate the GitHub login process. Please try again. (Discussion #17485)





 
Another possibility is that a proxy is needed on your system. The latest few releases contain technical changes, which don't fully support proxy servers yet.

**Reply by jansorg** [reply]
Unfortunately, there's no version supporting proxy servers yet.

**Comment by wwzhxbc** [comment]
Thank you for your reply.
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "github/feedback"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年5月30日(星期一) 下午3:55
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [github/feedback] Failed to initiate the GitHub login process. Please try again. (Discussion #17485)





 
Unfortunately, there's no version supporting proxy servers yet.

**Reply by HackerOQ** [reply]
Hello, That happened to me, too. Did you fix it?

**Reply by RafaelNevesdeOliveira** [reply]
The same happen to me, this was fixed when I disabled the extension

**Comment by wwzhxbc** [comment]
No,I still failed to login.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "github/feedback"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年5月31日(星期二) 上午10:36
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [github/feedback] Failed to initiate the GitHub login process. Please try again. (Discussion #17485)





 
Hello, That happened to me, too. Did you fix it?

**Comment by wwzhxbc** [comment]
Which city do you live in? Is it the copilot plug-in problem?
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "github/feedback"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年5月30日(星期一) 晚上8:23
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [github/feedback] Failed to initiate the GitHub login process. Please try again. (Discussion #17485)





  
Is it possible that your employer, internet service provider, or government, may be intercepting your SSL traffic?
  
It should be that the network is restricted, but I still can't log in after opening the proxy. I think it should be a plug-in adaptation problem, as the brother said above.

**Comment by JJpZh** [comment]
me too

**Reply by michelyao** [reply]
me too

**Reply by michelyao** [reply]
pycharm:2022.1(professional edition) + copilot:1.1.23.1570

**Comment by rpollard-crestron** [comment]
I am seeing the same issue with Android Studio plugin

Android Studio Chipmunk | 2021.2.1 Patch 1
Build #AI-212.5712.43.2112.8609683, built on May 18, 2022
Runtime version: 11.0.12+7-b1504.28-7817840 amd64
VM: OpenJDK 64-Bit Server VM by Oracle Corporation
Windows 10 10.0
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 8
Registry: external.system.auto.import.disabled=true
Non-Bundled Plugins: wu.seal.tool.jsontokotlin (3.7.3), some.awesome (1.14), com.github.setial (4.0.2), name.kropp.intellij.makefile (212.5712.51), PlantUML integration (5.13.0), org.intellij.plugins.markdown (212.5457.16), org.jetbrains.kotlin (212-1.6.21-release-334-AS5457.46), izhangzhihao.rainbow.brackets (6.23), com.github.copilot (1.1.23.1570)

**Reply by rpollard-crestron** [reply]
FYI: After updating the plugin to the latest version, Copilot is working again

**Comment by wlnRes** [comment]
我的也这样

**Reply by wlnRes** [reply]
设置电脑dns成8.8.8.8就可以了

**Reply by lepollo** [reply]
设置dns有效！ 设置完成后cmd命令 ipconfig /flushdns启用新的dns

**Reply by wwzhxbc** [reply]
我设置dns并清理了dns缓存，咋还是不行

**Reply by wwzhxbc** [reply]
> 设置dns有效！ 设置完成后cmd命令 ipconfig /flushdns启用新的dns

你pycharm和copilot版本分别是多少

**Reply by sele988** [reply]
pycharm和copilot 都是最新的，今天下载的，尝试了8.8.8.8，也是不行呢

**Reply by cloudinnng** [reply]
> 设置电脑dns成8.8.8.8就可以了

thanks . you save me

**Reply by katorly** [reply]
> pycharm和copilot 都是最新的，今天下载的，尝试了8.8.8.8，也是不行呢

那么您可以尝试修改ipv6地址，ipv4上8.8.8.8对应的ipv6地址是2001:4860:4860::8888  
我就是ipv4改了没用，但ipv6就可以

**Reply by wwzhxbc** [reply]
> > pycharm和copilot 都是最新的，今天下载的，尝试了8.8.8.8，也是不行呢
> 
> 那么您可以尝试修改ipv6地址，ipv4上8.8.8.8对应的ipv6地址是2001:4860:4860::8888 我就是ipv4改了没用，但ipv6就可以

我修改IPV6也不行
![图片](https://user-images.githubusercontent.com/103719188/172162385-0066ba12-8ae0-477a-ab14-2c9eb9fd5f51.png)
![图片](https://user-images.githubusercontent.com/103719188/172162423-4222433b-e5eb-49dc-b0fc-3e652433644f.png)

**Reply by kite1smy** [reply]
我设置dns 可以了

**Reply by yuan1** [reply]
可以！

**Reply by xu-feiran** [reply]
not works

**Reply by zewailaa** [reply]
works for me, thx!

**Comment by pakchoily** [comment]
`hosts` 增加 api.github.com 的 ip,可以解决。https://github.com/github-community/community/discussions/16965#discussioncomment-2881851

**Reply by liicos** [reply]
goooooooooood!

**Reply by yagamiy1023** [reply]
大佬能具体一点吗，用vpn的能登录吗通过这个方法

**Reply by pakchoily** [reply]
> 大佬能具体一点吗，用vpn的能登录吗通过这个方法

就是参照 我给的解决方案 在hosts 指定网址的ip，开不开vpn都不影响。

**Comment by kotisyamala** [comment]
I too have the same issue with IntelliJ's Latest version.
`**_11:01 PM	GitHub Copilot: Failed to initiate the GitHub login process. Please try again._**`

**Comment by chouyang** [comment]
Same issue here. But it works during night time though. Can't figure out why. Does not work with VPN either.

**Reply by SergeyStroganoff** [reply]
I too have the same issue with IntelliJ's Latest version. copilot version 1.1.24.1610. Does not work with VPN either.

**Comment by joseserro** [comment]
I had this issue before and fixed it by downgrading to the last version that worked. It stopped working today because I was forced to update it.

It's working on my laptop but not my desktop. The only difference is that I'm on windows 7 on my desktop.

There are no weird network configurations on either machine, nor my physical network. I'm not being monitored and traffic is okay otherwise. Pinging api.github.com yields 140.82.121.6 at the time of writing.

I've tried the hosts fix and it didn't work.

My guess is that it could have to do with some weird expired certificate.

**Reply by dayu521** [reply]
#17961 ,The same issue

**Reply by Solotov** [reply]
A possible quick fix :

+ check that you have the latest version of copilot.
+ check that you are logged in to github.

In case of windows 10 there is a process named `copilot-agent-win.exe`,
that is not digitally signed, some third party firewalls,
do not allow unsigned executables to establish a remote connection as a default rule.


In this case you should add the executable "copilot-agent-win.exe" to whitelist.
In any case you should restart the IDE, to force the copilot agent to restart.


If after all these steps Github Copilot does not login, try using a VPN or a proxy on your system and restart the IDE.

I hope you found this fix helpful.

**Reply by zxk0365** [reply]
Hello ！I have successfully opened github website with FastGithub, but Copilot has been unable to connect to Github and log in to Github in idea. I have tried to put copilot in the computer whitelist, but Copilot still cannot connect to Github. I really don't know what is the problem.

**Comment by mfcallahan** [comment]
I am also experiencing this same issue in WebStorm 2022.1.2 Build #WS-221.5787.30 with the latest version of GitHub Copilot. I am on an enterprise VPN connection, however all other GitHub authorizations work fine.

![image](https://user-images.githubusercontent.com/3037284/172894245-c145262f-2112-4e83-81e5-bab9cb03df14.png)

**Comment by CreatorEdition** [comment]
I also had this issue after a recent update
The login authorization page can be opened through the browser
Plugin version 1.1.25.1678

**Comment by Drawmahs** [comment]
Same issue. The plugin worked fine, but after a while this error appeared and the plugin has not been working in Intellij IDEA for a week. Updates don't help.

**Reply by zxk0365** [reply]
I've updated it twice and it still doesn't work

**Reply by CreatorEdition** [reply]
> 我更新了两次还是不行

可以尝试根据恢复到旧版本，然后离线安装插件，现在的问题是因为新版本部分数据没有通过IDE中的代理来进行访问

1、卸载插件，重启编辑器
2、下载[历史版本 https://plugins.jetbrains.com/plugin/download?rel=true&updateId=172765
3、代码器->插件设置->设置->从磁盘安装插件->选择下载的 zip包

服用完成！
解决方案来源 【[https://github.com/orgs/github-community/discussions/16965#discussioncomment-2916107】](https://github.com/orgs/github-community/discussions/16965#discussioncomment-2916107%E3%80%91)

**Reply by Tchekda** [reply]
> https://plugins.jetbrains.com/plugin/download?rel=true&updateId=172765

The login worked for me ! 

_Edit: The copilot server failed and asked for update_

**Reply by CreatorEdition** [reply]
> The login worked for me !
> 
> _Edit: The copilot server failed and asked for update_

先把版本更新到最新版本，然后用离线插件包的 core-1.1.20 变更文件名称为 core-1.1.25，并覆盖到 
...\AppData\Roaming\JetBrains\XXXXXXXXXXXXXXX\plugins\github-copilot-intellij\lib
重启启用插件

**Reply by sscoo** [reply]
v2ray要在设置里打开允许局域网的连接！ttttt

**Comment by Tchekda** [comment]
Working solution : https://github.com/github-community/community/discussions/18132#discussioncomment-2918095

**Reply by tamersalama** [reply]
This solution worked for me. My organization has a self-signed certificate.

**Comment by c2ray** [comment]
It seems to be the wrong requesting url that copilot always failed to initiate the GitHub login process.
The requesting url should be https://github.com/login/device, but now it always requests to https://github.com/login/device/code.

**Comment by mu7mdmagdy** [comment]
Same issue

**Reply by CreatorEdition** [reply]
下载历史版本离线插件 https://plugins.jetbrains.com/plugin/download?rel=true&updateId=172765
先把版本更新到最新版本，然后用离线插件包的 core-1.1.20 变更文件名称为 core-1.1.25，并覆盖到
...\AppData\Roaming\JetBrains\XXXXXXXXXXXXXXX\plugins\github-copilot-intellij\lib
重启启用插件

**Comment by raulblanko** [comment]
Same issue

**Comment by Rand-Random** [comment]
Duplicate of: https://github.com/orgs/github-community/discussions/13979

**Comment by jusescn** [comment]
idea → help → edit vm property  add  -Dcopilot.agent.disabled=true 

登陆 ：https://github.com/settings/copilot  修改 Suggestions matching public code 选择 allow

**Comment by markuszeller** [comment]
Same with PHPStorm on Windows 11 and Ubuntu 20.
`Failed to initiate the GitHub login process. Please try again.`

Happens very often on weekends and lasts a long time!

Update:
A new **copilot-agent-win.exe** was deployed and blocked in the firewall.

**Reply by thihan200** [reply]
How you fixed this problem ?

**Reply by markuszeller** [reply]
My firewall was blocking copilot-agent-win.exe which was a new version. After granting access, it worked again.

**Reply by thihan200** [reply]
What is the file path of copilot-agent-win.exe ?

**Reply by markuszeller** [reply]
C:\Users\markus\AppData\Roaming\JetBrains\PhpStorm2022.2\plugins\github-copilot-intellij\copilot-agent\bin\copilot-agent-win.exe

**Reply by thihan200** [reply]
![Screenshot_337](https://user-images.githubusercontent.com/71422014/191520867-50bd0ca2-a2f5-44b6-9c1c-73c1159235a9.jpg)

I gave access to that file. but it's still not working.

**Comment by Moses-Code-Dev** [comment]
This work for me https://github.com/djoxy-code/Github-copilot-Fix

**Reply by hsuanyi-chou** [reply]
thanks! works for me, too!!

**Reply by thihan200** [reply]
Thank you so much.

**Reply by hsuanyi-chou** [reply]
I fork yours and upload those files. hope to be done easily: https://github.com/hsuanyi-chou/Github-copilot-Fix
thank you so much !!!

**Comment by Davidmartcab** [comment]
I have a MacBook air m1, Android Studio Dolphin | 2021.3.1, GitHub Copilot 1.1.33.1980 I just tried: 1.1.31, 1.1.31, 1.1.24 and none of them  works.
All the times I have the same error: "The login to GitHub failed. Please try again."

**Reply by Moses-Code-Dev** [reply]
https://github.com/djoxy-code/Github-copilot-Fix

**Reply by santoshvarma4u** [reply]
Thank you

**Comment by micky-meecky** [comment]
我的解决过程是，开代理，把hosts的ip全部删掉，即可。之前我开着加速，也改了hosts，下了无数个版本的copilot，结果都不行，后来索性把hosts全删了，结果以下就可以了。

**Reply by qq2221629520** [reply]
妙啊  我也试了一下 就登上了
